File: BlueTeam/api/schemas/incident.py
from datetime import datetime
from pydantic import BaseModel


class IncidentResponse(BaseModel):
    incident_id: str
    alert_id: str
    event_id: str
    created_at: datetime
    updated_at: datetime
    severity: str
    risk_score: int
    attack_types: list[str]
    source_ips: list[str]
    rule_names: list[str]
    detection_count: int
    reasons: list[str]
    status: str


class IncidentListResponse(BaseModel):
    """
    Response model for GET /incidents.
    """
    persisted: bool = True
    status: str = "OK"
    message: str = "Incidents are successfully persisted in the database."
    incidents: list[IncidentResponse] | None = None

File: BlueTeam/dashboard/src/types/incident.ts
export interface Incident {
  incident_id: string;
  created_at: string;
  updated_at: string;
  alert_id: string;
  severity: string;
  risk_score: number;
  attack_types: string[];
  source_ips: string[];
  rule_names: string[];
  detection_count: number;
  reasons: string[];
  status: 'OPEN' | 'INVESTIGATING' | 'RESOLVED' | string;
}

export interface IncidentListResponse {
  persisted: boolean;
  status: string;
  message: string;
  incidents: Incident[] | null;
}
